### PR TITLE
Feature user definable product list

### LINKED ISSUE #186: Add user definable product list

### PR DESCRIPTION
**Pull request by sikai00**
Closes #186.

### CODE REVIEW

File: src/main/java/seedu/address/model/client/ClientRebuilder.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        return this;
+    }
+    /**
+     * Sets the {@code Set<Product>} of the {@code Client} that we are building.
+     */
+    public ClientRebuilder withProducts(Set<Product> products) {
+        this.products = products;
+        return this;
+    }
+
```
Comment (ThomasHoooo) [inline src/main/java/seedu/address/model/client/ClientRebuilder.java:76]: I believe that you can add one more method `withBirthday`?

**Comment by ThomasHoooo** [comment]
Just did a smoke test and tested all commands, all seem to work fine

**Review by ThomasHoooo — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/model/client/ClientRebuilder.java (hunk 1 above)
Comment (sikai00) [reply src/main/java/seedu/address/model/client/ClientRebuilder.java:86]: Added in merge commit

**Review by Thing1Thing2 — APPROVED** [review]

**Pull request merged**